model/model.go

```diff
@@ -420,22 +420,50 @@ func (m *Model) subtract(newModel *Model) *diff.Changes {
 
 	if newModel == nil {
 		for objName := range m.Objects {
-			chgs.Objects = append(chgs.Objects, string(objName))
+			chgs.Objects = append(chgs.Objects, objName)
 		}
 		return chgs
 	}
 
 	for objName, obj := range m.Objects {
 		if newModel.Objects[objName] == nil {
-			chgs.Objects = append(chgs.Objects, string(objName))
+			chgs.Objects = append(chgs.Objects, objName)
 		} else {
-			for relname := range obj.Relations {
+			for relname, rel := range obj.Relations {
 				if newModel.Objects[objName].Relations[relname] == nil {
-					chgs.Relations[string(objName)] = append(chgs.Relations[string(objName)], string(relname))
+					if chgs.Relations[objName] == nil {
+						chgs.Relations[objName] = make(map[types.RelationName][]string, 0)
+					}
+					chgs.Relations[objName][relname] = []string{}
+					continue
+				}
+				relDiff := substractArray(rel.Union, newModel.Objects[objName].Relations[relname].Union)
+				if len(relDiff) > 0 {
+					if chgs.Relations[objName] == nil {
+						chgs.Relations[objName] = make(map[types.RelationName][]string, 0)
+					}
+					chgs.Relations[objName][relname] = relDiff
 				}
 			}
 		}
 	}
 
 	return chgs
 }
+
+func substractArray(arr1, arr2 []*types.RelationRef) []string {
+	result := []string{}
+	for _, elem := range arr1 {
+		found := false
+		for _, elem2 := range arr2 {
+			if elem == elem2 {
```

Comment:
I don't think this would work since elem and elem2 are both pointers. They'll only be equal if they pointed at the same object instance. This should be `if *elem == *elem2 {`.
This suggests that the tests don't really cover this case and we need better test coverage.

Reply:
We have tests for this, and it works for this case without dereferencing pointers https://go.dev/play/p/i2VomqeWnh7?v=goprev  yields the same response as https://go.dev/play/p/tfJFSrVs-CJ?v=goprev